api/user.js

```diff
@@ -0,0 +1,52 @@
+import { BAD_REQUEST, TYPE_JSON } from './defs.js'
+import { POST } from './router.js'
+import * as db from './db.js'
+
+// function to check the form's data
+const verifyInformations = (body) => {
+  let checks = []
+  for (const [name, item] of Object.entries(body)) {
+    switch (name) {
+      case 'own_contact':
+        isNaN(item) && checks.push('please enter your phone number')
+        break
+      case 'emergency_contact':
+        isNaN(item) &&
+          checks.push(
+            'please enter a phone number to join in case of emergency',
+          )
+        break
+      default:
+        break
+    }
+  }
+  return checks
+}
+
+POST.user.registerForm = async ({ session, request }) => {
+  const contentType = request.headers.get('content-type') || ''
+  if (!contentType.includes('form')) {
+    return new Response('Bad request', BAD_REQUEST)
+  } else {
+    const body = await request.json()
+    const errors = verifyInformations(body)
```

Comment:
the way you did it now, I would have written: ```js
const errors = []
/some-test/.test(body.ownContact) || errors.push(['ownContact', 'please enter your phone number'])
/some-test/.test(body.emergencyContact) || errors.push([
  'emergencyContact',
  'please enter a phone number to join in case of emergency',
])
if (errors.length > 0) {
  return new Response(
    { errors: Object.fromEntries(errors) },
    { status: 200, headers: TYPE_JSON },
  )
}
```

Reply:
But it make sense to have some kind of generic function to test this, in that case I would write some validation system that would could be used like so:
```js
POST.user.registerForm = withBody(async ({ session, body }) => {
  // handle the request with the body already validated...
}, {
  // define how to validate the body here
  ownContact: value => /* js code to test that the value is ok*/,
  emergencyContact: value => /* js code to test that the value is ok*/,
})
```
I might let the client write the proper strings and just return the error message of the test for each field